ludwig/models/gbm.py

```diff
@@ -86,6 +88,8 @@ def compile(self):
             else {}
         )
         self.compiled_model = convert(self.lgbm_model, "torch", extra_config=extra_config)
+        yield
+        self.compiled_model = None
```

Comment:
This should be wrapped in a try-finally (in case an exception is raised, we still set back to `None`).